code-conventions.md

```diff
@@ -75,16 +114,19 @@ code. Contributors should also see the dedicated
   - Effect: init ->  execute  -> toString
   - Condition: init -> check -> toString
 
-## Naming
-* Classes named in CapitalCamelCase
-* Fields and methods named in camelCase
-  - Final static fields may be named in SCREAMING_SNAKE_CASE
-* Localized messages should be named like m_message_name
+
+### Naming
+* Class names are written in `UpperCamelCase`
+  - The file name should match its primary class name (e.g. `MyClass` goes in `MyClass.java`.)
+* Fields and methods named in `camelCase`
+  - Static constant fields should be named in `UPPER_SNAKE_CASE`
```

Comment:
This change will likely require code changes (constant requirement instead of just static final)

Reply:
I left this slightly ambiguous because there's actually a big discrepancy about what a "constant" actually is between the major style guides - I'm absolutely fine with continuing with our current interpretation or just going on a case-by-case basis.

Reply:
I'm personally in favor of requiring actual constants for this (constants being final primitives or final objects whose (associated) data can not change over time).

So I think the change is fine, it's just that there are likely uses of upper snake case in Skript fields which do not adhere to this stricter condition, but I think we can leave them and remove as they're touched (like we do with final on parameters and variables).

Reply:
Sounds good to me :)